server/routes/ad.ts

```diff
@@ -166,12 +166,36 @@ const getDetector = async (
     const response = await callWithRequest(req, 'ad.getDetector', {
       detectorId,
     });
+    let detectorState;
+    try {
+      const detectorStateResp = await callWithRequest(
+        req,
+        'ad.detectorProfile',
+        {
+          detectorId: detectorId,
+        }
+      );
+
+      const detectorStates = getFinalDetectorStates(
+        [detectorStateResp],
+        [convertDetectorKeysToCamelCase(response.anomaly_detector)]
+      );
+      detectorState = detectorStates[0];
+    } catch (err) {
+      console.log('Anomaly detector - Unable to retrieve detector state', err);
+    }
     const resp = {
       ...response.anomaly_detector,
       id: response._id,
       primaryTerm: response._primary_term,
       seqNo: response._seq_no,
       adJob: { ...response.anomaly_detector_job },
+      //@ts-ignore
+      ...(detectorState !== undefined ? { curState: detectorState.state } : {}),
+      ...(detectorState !== undefined
+        ? //@ts-ignore
+          { initializationError: detectorState.error }
```

Comment:
maybe only assign `initializationError` when the detector state is `DETECTOR_STATE.INIT_FAILURE`. I'm just thinking that with the current code could assign an unexpected failure error to `initializationError`. Although I see this is a small thing since the state is handled in the switch statement in `DetectorStateDetails`

Reply:
I guessed it is okay to assign `error` to `initializationError `. During initializing, `detector.error` does return message like "No full shingle in current detection window" or "No data in current detection window", which can be helpful if we want to show details about initialization.

Reply:
Yeah, I see. Actually, that probably makes more sense. Sounds good to leave